fuse_publishers/src/pose_2d_publisher.cpp

```diff
@@ -105,29 +109,58 @@ namespace fuse_publishers
 {
 
 Pose2DPublisher::Pose2DPublisher() :
-  fuse_core::AsyncPublisher(1),
+  fuse_core::AsyncPublisher(),
   device_id_(fuse_core::uuid::NIL),
   publish_to_tf_(false),
+  tf_timeout_(rclcpp::Duration(0, 0)),
   use_tf_lookup_(false)
 {
 }
 
+void Pose2DPublisher::initialize(
+  fuse_core::node_interfaces::NodeInterfaces<
+    fuse_core::node_interfaces::Base,
+    fuse_core::node_interfaces::Clock,
+    fuse_core::node_interfaces::Logging,
+    fuse_core::node_interfaces::Parameters,
+    fuse_core::node_interfaces::Services,
+    fuse_core::node_interfaces::Timers,
+    fuse_core::node_interfaces::Topics,
+    fuse_core::node_interfaces::Waitables
+  > interfaces,
+  const std::string & name,
+  size_t thread_count)
+{
+  interfaces_ = interfaces;
+  fuse_core::AsyncPublisher::initialize(interfaces_, name, thread_count);
+}
+
 void Pose2DPublisher::onInit()
 {
+  logger_ = std::make_shared<rclcpp::Logger>(interfaces_.get_node_logging_interface()->get_logger());
```

Comment:
I'm not sure we gain anything from `std::shared_ptr` here. The [API is returning a copy of a logger](https://github.com/ros2/rclcpp/blob/3fb012e2e979475f5044ab0e0f9b91d336db5f46/rclcpp/include/rclcpp/node_interfaces/node_logging_interface.hpp#L45-L46), so we may as well `std::move` that copy into a member variable instead of creating a shared pointer to another copy.

Reply:
The logger default constructor is private :(
I'll just spin up a logger with the name "uninitialized" until its replaced

Reply:
https://github.com/locusrobotics/fuse/compare/f9d587a8c481bab1398f71dd2a4a5eb28fa82547..6ee89e8d6b24b282fb99c14482fe2d6412dcc3e9